sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md

```diff
@@ -81,23 +137,30 @@ export interface Secret {
 export class SecretClient {
     constructor(endPoint: string, credential: TokenCredential, pipelineOrOptions?: ServiceClientOptions | NewPipelineOptions);
     backupSecret(secretName: string, options?: RequestOptionsBase): Promise<Uint8Array | undefined>;
+    beginDeleteSecret(name: string, options?: SecretPollerOptions): Promise<DeleteSecretPoller>;
+    beginRecoverDeletedSecret(name: string, options?: SecretPollerOptions): Promise<RecoverDeletedSecretPoller>;
     protected readonly credential: TokenCredential;
-    deleteSecret(secretName: string, options?: RequestOptionsBase): Promise<DeletedSecret>;
     static getDefaultPipeline(credential: TokenCredential, pipelineOptions?: NewPipelineOptions): ServiceClientOptions;
     getDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<DeletedSecret>;
     getSecret(secretName: string, options?: GetSecretOptions): Promise<Secret>;
```

Comment:
Should return a `Promise<KeyVaultSecret>`. In fact, I didn't think to check that with #5609 (`Key`->`KeyVaultKey`) so please check that, too.

Reply:
Addressing this here: https://github.com/Azure/azure-sdk-for-js/issues/5577